### PR TITLE
#561 added languages tweets

### PR DESCRIPTION
**Pull request by krzyk**
I had to add suppress for PMD, because it wasn't counting correctly the amount of `StringBuilder` reserved (PMD stated that builder was initialized with 16 capacity, while it is with 200)

### CODE REVIEW

**Comment by alex-palevsky** [comment]
Many thanks, I'll find someone to review it before we merge

**Comment by alex-palevsky** [comment]
@longtimeago review this pull request when possible

File: src/test/java/com/rultor/agents/twitter/TweetsTest.java [hunk 1]
```diff
@@ -60,6 +74,78 @@ public void postsTweet() throws Exception {
...
+     */
+    @Test
+    public void postsTweetWithLanguages() throws Exception {
+        final Repo repo = this.repo();
+        final List<Language> langs = Lists.newArrayList(
+            this.lang("Java"), this.lang("Python")
+        );
+        Mockito.when(repo.languages()).thenReturn(langs);
+        final Twitter twitter = Mockito.mock(Twitter.class);
+        final Agent agent = new Tweets(repo.github(), twitter);
```
Comment (longtimeago) [inline src/test/java/com/rultor/agents/twitter/TweetsTest.java]: looks like one-time variable

File: src/test/java/com/rultor/agents/twitter/TweetsTest.java [hunk 2]
```diff
@@ -60,6 +74,78 @@ public void postsTweet() throws Exception {
...
+    }
+
+    /**
+     * Tweets can post a tweet with language tags.
+     * @throws Exception In case of error.
+     */
+    @Test
+    public void postsTweetWithLanguages() throws Exception {
+        final Repo repo = this.repo();
+        final List<Language> langs = Lists.newArrayList(
```
Comment (longtimeago) [inline src/test/java/com/rultor/agents/twitter/TweetsTest.java]: we don't need mutability here. Consider using ```ImmutableList.of()```

File: src/test/java/com/rultor/agents/twitter/TweetsTest.java [hunk 3]
```diff
@@ -60,6 +74,78 @@ public void postsTweet() throws Exception {
...
+            )
+        );
+    }
+
+    /**
+     * Create mock issue.
+     * @param repo Repo to use
+     * @return Mocked issue.
+     */
+    private Issue issue(final Repo repo) {
```
Comment (longtimeago) [inline src/test/java/com/rultor/agents/twitter/TweetsTest.java]: why not static?

File: src/test/java/com/rultor/agents/twitter/TweetsTest.java [hunk 4]
```diff
@@ -60,6 +74,78 @@ public void postsTweet() throws Exception {
...
+        Mockito.when(repo.issues()).thenReturn(issues);
+        return issue;
+    }
+
+    /**
+     * Create mock repo.
+     * @return Mocked repo
+     * @throws IOException In case of error
+     */
+    private Repo repo() throws IOException {
```
Comment (longtimeago) [inline src/test/java/com/rultor/agents/twitter/TweetsTest.java]: same here

File: src/test/java/com/rultor/agents/twitter/TweetsTest.java [hunk 5]
```diff
@@ -73,10 +159,24 @@ public void postsTweet() throws Exception {
...
-        );
+        return talk;
     }
 
+    /**
+     * Create mock language.
+     * @param name Name of the language
+     * @return Language created.
+     */
+    private Language lang(final String name) {
```
Comment (longtimeago) [inline src/test/java/com/rultor/agents/twitter/TweetsTest.java]: and here

File: src/test/java/com/rultor/agents/twitter/TweetsTest.java [hunk 6]
```diff
@@ -46,6 +56,10 @@
  * @author Yegor Bugayenko (yegor@tpc2.com)
  * @version $Id$
  * @since 1.30
+ * @todo #561 When all the puzzles from
```
Comment (longtimeago) [inline src/test/java/com/rultor/agents/twitter/TweetsTest.java]: actually, https://github.com/jcabi/jcabi-github/issues/923 is already implemented

File: src/test/java/com/rultor/agents/twitter/TweetsTest.java [hunk 7]
```diff
@@ -46,6 +56,10 @@
...
+ *  https://github.com/jcabi/jcabi-github/issues/923 are implemented, remove
```
Comment (longtimeago) [inline src/test/java/com/rultor/agents/twitter/TweetsTest.java]: actually, jcabi/jcabi-github#923 is already implemented

**Comment by longtimeago** [comment]
@krzyk Please, see several remarks in the code.
Actually, actually, jcabi/jcabi-github#923 is already implemented, so we can start using ```MkGithub``` from this issue (if there is no other blocker)

**Comment by krzyk** [comment]
@longtimeago jcabi/jcabi-github#923 was implemented but not all the puzzles, right now the code is in a state where MkGithub languages throws unimplemented exception

**Comment by krzyk** [comment]
@longtimeago I took another look and the code is here after all, something changed during the weekend

**Comment by krzyk** [comment]
@yegor256 could you release new version of jcabi-github? This way I could get rid of todo here and have full implementation ready.

**Comment by yegor256** [comment]
@krzyk jcabi-github 0.18.9 will be released in a few minutes, see https://github.com/jcabi/jcabi-github/issues/972

**Comment by krzyk** [comment]
@longtimeago take a look now, with newer jcabi-github it is much cleaner :)

**Comment by longtimeago** [comment]
@krzyk totally agree, thank you!

**Comment by longtimeago** [comment]
@rultor good to merge

**Comment by yegor256** [comment]
@rultor merge

**Comment by alex-palevsky** [comment]
@longtimeago Much obliged! I've added **16 mins** to your account in payment "50957905". +16 to your rating, your total score is [+1514](http://www.netbout.com/b/35024?open=rating)

**Comment by alex-palevsky** [comment]
@rultor deploy now pls

**Pull request merged**